TYPO3.Neos/Classes/TYPO3/Neos/Service/Mapping/NodePropertyConverterService.php

```diff
@@ -0,0 +1,248 @@
+<?php
+namespace TYPO3\Neos\Service\Mapping;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "TYPO3.Neos".            *
+ *                                                                        *
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU General Public License, either version 3 of the   *
+ * License, or (at your option) any later version.                        *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\Flow\Property\PropertyMapper;
+use TYPO3\Flow\Property\PropertyMappingConfiguration;
+use TYPO3\Flow\Property\PropertyMappingConfigurationInterface;
+use TYPO3\Flow\Reflection\ObjectAccess;
+use TYPO3\Flow\Utility\TypeHandling;
+use TYPO3\TYPO3CR\Domain\Model\NodeInterface;
+use TYPO3\TYPO3CR\Domain\Model\NodeType;
+
+/**
+ * Creates PropertyMappingConfigurations to map NodeType properties for the Neos interface.
+ *
+ * @Flow\Scope("singleton")
+ */
+class NodePropertyConverterService
+{
+    /**
+     * @Flow\InjectConfiguration(package="TYPO3.Neos", path="userInterface.inspector.dataTypes")
+     * @var array
+     */
+    protected $typesConfiguration;
+
+    /**
+     * @Flow\Inject
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * @Flow\Inject
+     * @var PropertyMapper
+     */
+    protected $propertyMapper;
+
+    /**
+     * @Flow\Transient
+     * @var array
+     */
+    protected $generatedPropertyMappingConfigurations = array();
+
+    /**
+     * Get a single property reduced to a simple type representation
+     *
+     * @param NodeInterface $node
+     * @param $propertyName
+     * @return string
+     */
+    public function getProperty(NodeInterface $node, $propertyName)
+    {
+        return $this->getPropertyInternal($node, $propertyName);
+    }
+
+    /**
+     * Get all properties as JSON encoded string representation
+     *
+     * @param NodeInterface $node
+     * @return string
+     */
+    public function getPropertiesJson(NodeInterface $node)
+    {
+        $properties = array();
+        foreach ($node->getNodeType()->getProperties() as $propertyName => $propertyConfiguration) {
+            if ($propertyName[0] === '_' && $propertyName[1] === '_') {
+                // skip fully-private properties
+                continue;
+            }
+
+            $properties[$propertyName] = $this->getPropertyInternal($node, $propertyName);
+        }
+
+        return json_encode($properties);
```

Comment:
Wait, no, because we should only transfer configured properties of the node type. I think that was the catch. At least current behavior.